### DISCUSSION TITLE
Where is the leaderbord stored?

### DISCUSSION THREAD

**Discussion by Just-Nils** [EssentialsX/Essentials#4091] [Q&A] [answered]
I want to make a Discord bot, wich reads the leaderbord and returns it on command. To make this possible I need to know, how I (or the bot) can read the leaderbord.

**Comment by triagonal** [comment, marked as answer]
Are you talking about the Top Balances leaderboard (`/baltop`)? If so, you can use the new `BalanceTop` API service. You can access it using the Bukkit ServicesManager as detailed [here](https://github.com/EssentialsX/Essentials/pull/3702#issue-497307512):
```java
import net.essentialsx.api.v2.services.BalanceTop;
//...
BalanceTop api = Bukkit.getServer().getServicesManager().load(BalanceTop.class);
```
Take a look at the [API interface](https://github.com/EssentialsX/Essentials/blob/2.x/Essentials/src/main/java/net/essentialsx/api/v2/services/BalanceTop.java) to see what kind of data you can access. Keep in mind that this service is only available on development builds, so ensure you're using the [latest dev build](https://essentialsx.net/downloads.html) of EssentialsX.

**Comment by mdcfe** [comment]
If by "leaderboard" you mean `/balancetop`, you can use the [`BalanceTop`](https://jd-v2.essentialsx.net/net/essentialsx/api/v2/services/balancetop) service to access the entries. Baltop data is calculated while the server is running and stored in memory, so if you want to access it from a Discord bot running in a separate process, you'll need to write a custom plugin as a bridge between your server and your bot.
